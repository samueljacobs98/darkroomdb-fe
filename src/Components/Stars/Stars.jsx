import "./Stars.scss";
import starFull from "../../assets/svgs/stars/full.svg";
import starHalf from "../../assets/svgs/stars/half.svg";
import starEmpty from "../../assets/svgs/stars/empty.svg";

const Stars = ({ rating }) => {
  const stars = {
    full: Math.floor(rating),
    half: Math.ceil(rating) - Math.floor(rating),
    empty: 5 - Math.ceil(rating),
  };

  const getStarJSX = (i) => {
    if (i < stars.full) return <img src={starFull} alt="full rating star" />;
    if (i < stars.full + stars.half)
      return <img src={starHalf} alt="half full rating star" />;
    return <img src={starEmpty} alt="full rating star" />;
  };

  return (
    <div className="stars">
      {getStarJSX(1)}
      {getStarJSX(2)}
      {getStarJSX(3)}
      {getStarJSX(4)}
      {getStarJSX(5)}
    </div>
  );
};

export default Stars;
