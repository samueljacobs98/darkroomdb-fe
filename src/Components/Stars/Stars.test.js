import { render } from "@testing-library/react";
import Stars from "./Stars";

it("should match component snapshot", () => {
  const { container } = render(<Stars rating={1} />);

  expect(container).toMatchSnapshot();
});
