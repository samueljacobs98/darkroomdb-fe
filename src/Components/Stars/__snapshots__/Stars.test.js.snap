// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match component snapshot 1`] = `
<div>
  <div
    class="stars"
  >
    <img
      alt="full rating star"
      src="empty.svg"
    />
    <img
      alt="full rating star"
      src="empty.svg"
    />
    <img
      alt="full rating star"
      src="empty.svg"
    />
    <img
      alt="full rating star"
      src="empty.svg"
    />
    <img
      alt="full rating star"
      src="empty.svg"
    />
  </div>
</div>
`;
